lib/authorize.js

```diff
@@ -0,0 +1,55 @@
+import { getLogger } from 'appium-logger';
+import { exec } from 'teen_process';
+import { getPath } from 'appium-xcode';
+import path from 'path';
+
+export async function authorize(insecure) {
+
+  let logger = getLogger('authorize');
+  let xcodeDir;
+  let user;
+
+  try { 
+    // enable developer tools
+    logger.info('Enabling DevToolsSecurity');
+    await exec('DevToolsSecurity --enable');
+
+    // update security db -- removes authorization prompt
+    logger.info(`Updating security db for ${insecure ? 'insecure' :
+        'developer'} access`);
+    let cmd = (`security authorizationdb write system.privilege.taskport ${insecure ? 'allow' : 'is-developer'}`);
+    await exec(cmd);
+    logger.info('Granting access to built-in simulator apps');
+
+    if (!process.env.HOME) {
+      logger.info(new Error('Could not determine your $HOME'));
```

Comment:
yes, we probably either want to throw this, or just log a string, rather than just log an error object